pkg/controller/cache/cache_controller.go

```diff
@@ -176,13 +177,23 @@ func (r *ReconcileCache) Reconcile(request reconcile.Request) (reconcile.Result,
 				}
 			}
 		}
+
+		ok, err := common.IsOwner(ispnInstance, instance, r.scheme)
```

Comment:
It seems that the `SetControllerReference` does some similar checks. Do we need to add our owns?

Reply:
Yes, `SetControllerReference` check for already set owner, but this validation prevents a redundant CR update